File: leetcode/1299.py
from typing import List

class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        
        largest_ele = arr[-1]
        arr[-1] = -1
        for idx in range(len(arr)-2,-1,-1):
            
            temp = arr[idx]

            arr[idx] = largest_ele
            if temp > largest_ele:
                largest_ele = temp
            

        return arr


nums = [17,15]

sol = Solution()
out = sol.replaceElements(nums)
print(out)

File: leetcode/arrays/1299.go
package main 
import "fmt"

func replaceElements(arr []int) []int {
    
	max_ele := arr[len(arr)-1]
	arr[len(arr)-1] = -1

	for iter:= len(arr)-2; iter>=0; iter--{
		temp := arr[iter]
		arr[iter] = max_ele
		if temp > max_ele{
			max_ele = temp
		}
	}

	return arr
}

func main(){

	arr := []int{400,300,500}

	ans := replaceElements(arr)

	fmt.Println(ans)
}